### PR TITLE
imxrt-flash: support for calulating CRC32 of selected region

### PR DESCRIPTION
**Pull request by ziemleszcz**
CRC32 can be calulated for selected region of flash memory or raw partition. To calculate automatically CRC32 of whole memory/partition set addr and len to 0.

JIRA: NIL-574

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imxrt1176).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: storage/imxrt-flash/flashsrv.c [hunk 1]
```diff
@@ -166,6 +168,54 @@ static int flashsrv_chipErase(uint8_t fID)
...
+	return crc32;
+}
+
+
+static int flashsrv_calcCrc32(uint8_t fID, size_t offset, size_t len, uint32_t *crc32)
+{
+	flashsrv_memory_t *mem;
+	uint8_t buf[CRC32_BUFSZ];
+	uint32_t tmp;
+	unsigned int i;
```
Comment (gorbanchovie) [inline storage/imxrt-flash/flashsrv.c:191]: pls fix

File: storage/imxrt-flash/imxrt-flashsrv.h [hunk 2]
```diff
@@ -22,14 +22,38 @@
...
+		} write;
+
+		/* directRead */
+		struct {
+			uint32_t addr;
+		} read;
+
+		/* calcCrc32 */
+		struct {
+			/* set addr & len to 0 for full range */
```
Comment (gorbanchovie) [inline storage/imxrt-flash/imxrt-flashsrv.h:52]: add init value

**Review by gorbanchovie — DISMISSED** [review]
lgtm

File: storage/imxrt-flash/flashsrv.c [hunk 3]
```diff
@@ -166,6 +168,62 @@ static int flashsrv_chipErase(uint8_t fID)
...
+	uint8_t *buf;
+	uint32_t tmp;
+	int res;
+
+	res = flashsrv_getFlashMemory(fID, &mem);
+	if (res < 0) {
+		return res;
+	}
+
+	buf = malloc(CRC32_BUFSZ);
```
Comment (gerard5) [inline storage/imxrt-flash/flashsrv.c:198]: The imxrt-flash is a legally relevant part of fw. All allocated memory should be known at initialization stage or compile time. Using malloc/free at each call to function may lead to memory fragmentation during long-term operation (MISRA Dir 4.12 req.).
Btw. NOR page size is 256 bytes @ 4KiB sector size, you can use smaller buffer for it e.g. of page size 256 bytes if allocated statically to save ram space.

**Review by gorbanchovie — APPROVED** [review]
lgtm

**Pull request merged**